src/charm.py

```diff
@@ -22,26 +23,38 @@ class Charm(ops.CharmBase):
     """
 
     def __init__(self, *args: typing.Any):
-        """Charm the service.
-
-        Arguments:
-            args: Arguments passed to the CharmBase parent constructor.
-        """
+        """Charm the service."""
         super().__init__(*args)
-        self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-
-    def _on_install(self, _: ops.InstallEvent) -> None:
-        """Handle install event."""
-        logger.info("Installing Falco")
-        self.unit.status = ops.MaintenanceStatus("Installing Falco")
-        # TODO: Add Falco installation logic
-        self.unit.status = ops.ActiveStatus()
 
-    def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
-        """Handle config changed event."""
-        logger.info("Configuring Falco")
-        # TODO: Add Falco configuration logic
+        self.falco_layout = FalcoLayout(base_dir=self.charm_dir / "falco")
+        self.falco_service_file = FalcoServiceFile(self.falco_layout)
+        self.managed_falco_config = FalcoConfig(self.falco_layout, self)
+        self.falco_service = FalcoService(self.managed_falco_config, self.falco_service_file)
+
+        self.framework.observe(self.on.remove, self._on_remove)
+        self.framework.observe(self.on.install, self._on_install_or_upgrade)
+        self.framework.observe(self.on.upgrade_charm, self._on_install_or_upgrade)
+        self.framework.observe(self.on.update_status, self._update_status)
```

Comment:
It is my understanding that this handler can be removed

Reply:
We need this handler to act as a health check mechanism